src/backend/tcop/utility.c

```diff
@@ -680,7 +680,14 @@ standard_ProcessUtility(PlannedStmt *pstmt,
 						break;
 
 					case TRANS_STMT_PREPARE:
-						if (Gp_role == GP_ROLE_DISPATCH)
+						/*
+						 * SINGLENODE_FIXME:
+						 * It seems like `PREPARE TRANSACTION` works in singlenode mode,
+						 * but currently we just disable it to make it behaves closer
+						 * to cluster mode.
+						 * Maybe reconsider it in the future.
+						 */
+						if (Gp_role == GP_ROLE_DISPATCH || Gp_role == GP_ROLE_SINGLENODE)
```

Comment:
why not support `TRANS_STMT_PREPARE ` ?

Reply:
I'm not very familliar with this part code. Not very confident that it works correctly.
Since CBDB not support it, I think it is acceptable that we don't support it in singlenode mode.
It will also cause some of the tests behave more differently. 
How about let some one be famiiliar with this part of code to support it in the future?